RELEASE.md

```diff
@@ -1,5 +1,9 @@
 # Creating a Release
 
+Our release strategy is:
+
+> Release major/minor versions with detailed summaries when necessary, while releasing point/patch versions weekly or as needed without extensive announcements.
```

Comment:
nit: wrap at 100 chars

less nit: Our versioning scheme is 0.x.y where x is the major version and y is the minor, so we don't really have patch versions.